File: backend/src/main/java/com/prediccion/apppredicciongm/gestion_inventario/inventario/dto/response/InventarioResponse.java
package com.prediccion.apppredicciongm.gestion_inventario.inventario.dto.response;

import com.prediccion.apppredicciongm.enums.EstadoInventario;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class InventarioResponse {
    
    private Integer inventarioId;
    private Integer productoId;
    private String nombreProducto;
    private String categoriaNombre;
    private String unidadMedida;
    private Integer stockDisponible;
    private Integer stockReservado;
    private Integer stockEnTransito;
    private Integer stockTotal;
    private Integer stockMinimo;
    private Integer stockMaximo;
    private Integer puntoReorden;
    private String ubicacionAlmacen;
    private LocalDateTime fechaUltimoMovimiento;
    private LocalDateTime fechaUltimaActualizacion;
    private Integer diasSinVenta;
    private EstadoInventario estado;
    private Boolean necesitaReorden;
    private Boolean bajoPuntoMinimo;
    private String observaciones;
}

File: frontend/src/app/features/inventario/model/InventarioResponse.ts
import { Page } from "../../../shared/models/Page";


export interface InventarioResponse {
    inventarioId: number;
    productoId: number;
    nombreProducto: string;
    stockDisponible: number;
    stockReservado?: number;
    stockEnTransito?: number;
    stockMinimo: number;
    stockMaximo?: number;
    puntoReorden: number;
    ubicacionAlmacen: string;
    estado: string;
    observaciones?: string;
}


export interface PageInventarioResponse extends Page<InventarioResponse> { }